src/qfusion.cpp

```diff
@@ -64,39 +64,52 @@ QFusion::QFusion(QInterfacePtr target)
  * buffered (and doesn't "commute") and before output from qubits. The rest of the engine simply wraps the other public
  * methods of QInterface to flush or discard the buffers as necessary.
  */
+
+void QFusion::EraseControls(std::vector<bitLenInt> controls, bitLenInt qubitIndex)
+{
+    std::set<bitLenInt>::iterator found;
+    bitLenInt control;
+    for (bitLenInt i = 0; i < controls.size(); i++) {
+        control = controls[i];
+        found = bitControls[control].find(qubitIndex);
+        if (found != bitControls[control].end()) {
+            bitControls[control].erase(found);
+        }
+    }
+}
+
 void QFusion::FlushBit(const bitLenInt& qubitIndex)
 {
+    BitBufferPtr bfr = bitBuffers[qubitIndex];
+
     // If we ended up with a buffer that's (approximately or exactly) equal to identity operator, we can discard it
     // instead of applying it.
-    if (bitBuffers[qubitIndex] && bitBuffers[qubitIndex]->IsIdentity()) {
+    if (bfr && bfr->IsIdentity()) {
```

Comment:
Replace this and the subsequent if statement with an inversion - `if (!bfr) return;`, which simplifies and de-indents the rest of the code in this function.

Also apply this idiom to subsequent functions.

Reply:
Done.